### PR TITLE
Use numpy numpy arrays in results, instead of JAX.numpy arrays

### PR DESCRIPTION
**Pull request by InnocentBug**
For the final result, we should prefer using normal numpy arrays instead of JAXarrays.
This should help a lot with pickling the final results.
JAX arrays can be a bit tricky to reconstruct in envs, that are not the same as the simulations were run.

So moving to numpy arrays, should enable user to pickle final results on the cluster and do the analysis (like plotting) on a local machine.

### CODE REVIEW

File: pysages/methods/utils.py [hunk 1]
```diff
@@ -174,3 +183,35 @@ def listify(arg, replicas, name, dtype):
...
+
+    new_dict = {}
+    for key in dictionary:
+        if not numpy_only:
+            new_dict[key] = dictionary[key]
+        if isinstance(dictionary[key], str):
+            numpy_array = numpy.asarray(dictionary[key])
+            if numpy_array.dtype != numpy.dtype("O"):
+                new_dict[key] = numpy_array
+    return new_dict
```
Comment (pabloferz) [inline pysages/methods/utils.py:217]: I would write it like this:
```python
from pysages.utils import JaxArray

def isarray(val):
    return isinstance(val, JaxArray) or isinstance(val, numpy.ndarray)

def numpyfy_vals(dictionary: dict, arrays_only: bool = False):
    new_dict = copy.copy(dictionary) if arrays_only else {}
    for key, val in dictionary.items():
        if isarray(val):
            new_dict[key] = numpy.asarray(val)
```

Comment (InnocentBug) [reply pysages/methods/utils.py]: done

**Review by pabloferz — DISMISSED** [review]

**Comment by InnocentBug** [comment]
Sorry, I needed to revert your suggestion because of nested arrays.

**Review by pabloferz — APPROVED** [review]
It feels suboptimal trying to convert first to numpy arrays first, but let merge this for now and improve it later if possible.

**Pull request merged**